### PR TITLE
IR-471: Removing featuregate for chunkSizeMiB config

### PR DESCRIPTION
**Pull request by deepsm007**
https://github.com/openshift/api/pull/1948

/cc @flavianmissi 

/hold 

[Sippy test runs](https://sippy.dptools.openshift.org/sippy-ng/tests/4.17?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522never-stable%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522aggregated%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Chunksizemib%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&sort=asc&sortField=current_working_percentage)

Tested with cluster-bot using [this](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/release-openshift-origin-installer-launch-aws-modern/1819091147578937344)

```
cluster-bot]$ oc get clusterversion version
NAME      VERSION                                                AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.17.0-0.test-2024-08-01-193338-ci-ln-xxm7wdb-latest   True        False         5m24s   Cluster version is 4.17.0-0.test-2024-08-01-193338-ci-ln-xxm7wdb-latest

cluster-bot]$ oc patch config.image/cluster -p '{"spec":{"storage":{"s3":{"chunkSizeMiB":5}}}}' --type=merge
config.imageregistry.operator.openshift.io/cluster patched

cluster-bot]$ oc -n openshift-image-registry get  config.image/cluster -oyaml | yq .spec.storage.s3
bucket: ci-ln-xxm7wdb-76ef8-77pmb-image-registry-us-east-2-trhxwyqtpsk
chunkSizeMiB: 5
encrypt: true
region: us-east-2
trustedCA:
  name: ""
virtualHostedStyle: false
```

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -1834,3 +1847,4 @@ sigs.k8s.io/structured-merge-diff/v4/value
 ## explicit; go 1.12
 sigs.k8s.io/yaml
 # google.golang.org/grpc => google.golang.org/grpc v1.55.0
+# github.com/openshift/api v0.0.0-20240508125607-95e22923d553 => github.com/deepsm007/api v0.0.0-20240703125909-bbc85b7a2c96
```
Comment (deepsm007) [inline vendor/modules.txt:1850]: Only for testing

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-image-registry-operator
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.3
```
Comment (flavianmissi) [inline go.mod:5]: Was this automatically added? Does everything work fine without line 5?

**Review by flavianmissi — COMMENTED** [review]
Looking good.
Could you separate the go.mod, go.sum and vendor changes into its own commit?

File: go.mod (hunk 2 above)
Comment (deepsm007) [reply go.mod:5]: It was added with `go mod tidy && go mod vendor`. I can test by removing that line

**Comment by deepsm007** [comment]
> Could you separate the go.mod, go.sum and vendor changes into its own commit?

Yes, i'll update the PR

**Comment by flavianmissi** [comment]
left a nit but lgtm otherwise
/lgtm

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/cluster-image-registry-operator
 
-go 1.22
+go 1.22.0
```
[the remark this answers is not in the document]
Comment (deepsm007) [reply go.mod]: `make verify` is not happy with manual changes, Either it require toolchain or a specific version. Running `go mod tidy` updates the version and adds toolchain

**Comment by flavianmissi** [comment]
I'll give all the docs and px approval labels to make sure we can merge before 4.17 branching. @deepsm007 if I remember correctly we will not be writing user facing docs for this right? if that's the case then I think we don't need any action on docs/px side.
/label docs-approved
/label px-approved
I'll wait until the vendored reference for openshift/api gets updated (once https://github.com/openshift/api/pull/1948 merges) before giving the lgtm again.

**Comment by deepsm007** [comment]
> @deepsm007 if I remember correctly we will not be writing user facing docs for this right? 

This is going in as part of Default FeatureGate and an extension to docker registry configuration. I suggest we write some user facing docs.

**Comment by flavianmissi** [comment]
> This is going in as part of Default FeatureGate and an extension to docker registry configuration. I suggest we write some user facing docs.
Then please loop in @stevsmit, he's IR docs.

**Comment by deepsm007** [comment]
Failures are related to CI infra issue
/retest-required

**Pull request merged**